src/components/UsernameInput/UsernameInput.tsx

```diff
@@ -13,6 +12,19 @@ interface UsernameInputProps extends Omit<TextStyleProps, "onColorChange"> {
 }
 
 const UsernameInput: FC<UsernameInputProps> = ({ state, onInputChange, label }) => {
+    const [emailValidation, setEmailValidation] = useState<"valid" | "invalid">();
+    const emailRegex = new RegExp(/^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/);
+
+    useEffect(() => {
+        if (state !== undefined && state.variant === "email") {
+            validateEmail(state.textValue);
+        }
+    }, [state]);
```

Comment:
This `useEffect` will run whenever any of the state changes - i.e. whenever buttons or color changes, this one will also run - could we update the implementation to ensure it _**only runs**_ when the user enters new input ?

Reply:
Hello yes just modified the dependencies array to check only for the change of the textValue!